internal/api/schemas/v1.yaml

```diff
@@ -35,13 +186,99 @@ components:
       in: header
       name: X-Consul-Token
   schemas:
-    GatewayList:
+    GatewayPage:
       type: object
-      properties: {}
+      required:
+      - gateways
+      properties:
+        gateways:
+          type: array
+          items:
+            "$ref": "#/components/schemas/Gateway"
+          minItems: 0
+        meta:
+          type: object
+    Gateway:
+      type: object
+      required:
+      - name
+      - namespace
+      - listeners
+      properties:
+        name:
+          type: string
+          minLength: 1
+        namespace:
+          type: string
+          minLength: 1
+        listeners:
+          type: array
+          items:
+            "$ref": "#/components/schemas/Listener"
+          minItems: 1
+        meta:
+          type: object
+    Listener:
+      type: object
+      required:
+      - hostname
+      - port
+      - protocol
+      properties:
+        name:
+          type: string
+        hostname:
+          type: string
+        port:
+          type: number
+        protocol:
+          type: string
+          enum:
+          - http
+          - tcp
+        tls:
+          "$ref": "#/components/schemas/TLSConfiguration"
+    TLSConfiguration:
+      type: object
+      properties:
+        minVersion:
+          type: string
+        maxVersion:
+          type: string
+        cipherSuites:
+          type: array
+          items:
+            type: string
+        certificates:
+          type: array
+          items:
+            "$ref": "#/components/schemas/Certificate"
+    Certificate:
+      type: object
+      properties:
+        vault:
+          "$ref": "#/components/schemas/VaultCertificate"
+    VaultCertificate:
+      type: object
+      properties:
+        path:
+          type: string
+          minLength: 1
+        chainField:
+          type: string
+          minLength: 1
+        privateKeyField:
+          type: string
+          minLength: 1
     Error:
       type: object
       required:
       - message
+      - code
       properties:
+        code:
```

Comment:
Do you expect this to align with standard HTTP response codes or some custom set of codes? Helpful to add a description here IMO

Reply:
For now, yes, but I'm planning on modifying some of this stuff to be something more verbose and user-defined since you can already get this info via the HTTP response code, just haven't done it yet.